### DISCUSSION TITLE
Specifying all files with extension

### DISCUSSION THREAD

**Discussion by CaptainQuake** [samuel-lucas6/Kryptor#45] [Ideas]
You can add the function of encrypting and decrypting multiple files according to a template. 
Example: `kryptor -e *.jpg *.png`

and
`kryptor -e -r *.jpg *.png`

`-r` recursively for all subfolders

decrypt

`kryptor -d *.kryptor`
`kryptor -d -r *.kryptor`

**Comment by samuel-lucas6** [comment]
That's an interesting idea. Not sure how tricky that would be to implement.

However, `-r` is already taken, and I'd prefer to keep the number of options as small as possible. There are already quite a few due to signing as well as encryption and password, keyfile, and private/public key support. Too many and the program becomes GPG.

The other issue is that it wouldn't work the same with decryption because every encrypted file has the `.kryptor` extension. That's validated to save time.

**Reply by CaptainQuake** [reply]
To decrypt all files, you can use `kryptor -d -r *.kryptor`

It is very inconvenient to enter the name of each file for encryption and decryption.

**Reply by samuel-lucas6** [reply]
I can understand that, which is why there's folder encryption/decryption. However, that's going to be changed in the next version so the folder gets converted into a non-compressed ZIP and that file is encrypted.

**Comment by samuel-lucas6** [comment]
Ok I'm going to exclude this from the next release (v4.0.0), but I may revisit it down the line. The main thinking being:
- Just use directory encryption if you want to encrypt more than a few files. That's what it's there for.
- Is it just in the current directory or does it include subdirectories? It would probably require another option like you said, which then begs the question of whether you use the other option for directory encryption as well.
